### PR TITLE
[JUJU-2449] Allow building locally without a LXC container

### PR DESCRIPTION
**Pull request by SimonRichardson**
In jenkins we have complete control over the whole machine, so it seems very wasteful to put the build script in a container. It also makes it REALLY hard to debug if it's in a nested container.

## QA steps

```sh
$ dqlite-build
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,4 +30,5 @@ tests/tmp.*
 *.out
 *.charm
 *.bundle
-*.log
\ No newline at end of file
+*.log
+dqlite-deps.tar.bz2
```
Comment (beliaev-maksim) [inline .gitignore:34]:
```suggestion
dqlite-deps.tar.bz2

```

File: scripts/dqlite/build-script.sh [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+cp sqlite/.libs/*.a juju-dqlite-static-lib-deps/
+cp dqlite/.libs/*.a juju-dqlite-static-lib-deps/
+
+# Collect required headers
+mkdir juju-dqlite-static-lib-deps/include
+cp -r raft/include/* juju-dqlite-static-lib-deps/include
+cp -r sqlite/*.h juju-dqlite-static-lib-deps/include
+cp -r dqlite/include/* juju-dqlite-static-lib-deps/include
+
+tar cjvf juju-dqlite-static-lib-deps.tar.bz2 juju-dqlite-static-lib-deps
```
Comment (beliaev-maksim) [inline scripts/dqlite/build-script.sh:141]:
```suggestion
tar cjvf juju-dqlite-static-lib-deps.tar.bz2 juju-dqlite-static-lib-deps

```

**Review by beliaev-maksim — CHANGES_REQUESTED** [review]

**Review by manadart — COMMENTED** [review]
`make dqlite-build` works, but `make dqlite-local-build` results in:
```
make dqlite-local-build
DQLITE: Executing build script
bash: line 1: ./scripts/dqlite/build-script.sh: Permission denied
DQLITE: Placing lib archive
mv: cannot stat './scripts/dqlite/dqlite-deps.tar.bz2': No such file or directory
make: *** [scripts/dqlite/Makefile:65: dqlite-local-build] Error 1
```

**Review by manadart — APPROVED** [review]

**Pull request merged**